src/registry.rs

```diff
@@ -48,17 +48,35 @@ impl RegistryCore {
 
             self.dim_hashes_by_name.insert(desc.fq_name.clone(), desc.dim_hash);
 
-            desc.id
-        };
+            // If it is not a duplicate desc in this collector, add it to
+            // the collector_id.
+            if let None = id_set.iter().find(|id| **id == desc.id) {
+                id_set.push(desc.id);
+                collector_id += desc.id;
```

Comment:
L45 should be moved out of the loop and check `collector_id instead` of `desc.id`.